packages/svelte/src/lib/createInertiaApp.ts

```diff
@@ -67,24 +69,37 @@ export default async function createInertiaApp({
     if (progress) {
       setupProgress(progress)
     }
+  }
 
-    setup({
-      el,
-      App: SvelteApp,
-      props: {
-        initialPage,
-        resolveComponent,
-      },
-    })
+  let result = setup({
+    // @ts-expect-error
+    el,
+    App: !isServer ? App : SSR,
+    props: {
+      id,
+      initialPage,
+      resolveComponent,
+    },
+  })
 
-    return
-  }
+  if (isServer) {
+    if (!result && typeof (SSR as SSRComponentType).render !== 'function') {
```

Comment:
I think you can make a cleaner check using `import { VERSION } from 'svelte/compiler'`.

Reply:
Thanks for the follow up, I will need to double check that but I believe I had tried and this import is either not available il all svelte major versions or only available in the NodeJS context while compiling/transpiling with Vite. Also the import is going to fetch for Inertia's version and not the user specified version in his project.

At first I wanted to make this "transparent" to the user and dynamically render the proper way a component based on that version but I was unsuccessful 

https://stackoverflow.com/questions/78470116/are-dynamic-imports-at-runtime-in-the-browser-esm-possible-with-svelte